source/game/Game.gd

```diff
@@ -29,6 +30,17 @@ func _unhandled_input(event: InputEvent) -> void:
 			elif current_unit and not loc.unit:
 				current_unit.move_to(_get_path_for_unit(current_unit, loc))
 				_set_current_unit(null)
+			
+			elif (current_unit and loc.unit and loc.unit.side.number != current_side.number  
+				  and current_unit.reachable.has(loc)):
+				
+				current_unit.move_to(_get_path_for_unit(current_unit, loc))
+				yield(scenario, "unit_move_finished")
+				HUD.show_attack_popup(current_unit)
+				yield(HUD, "attack_selected")
```

Comment:
I'm pretty sure you can move `current_unit.execute_attack(loc.unit, current_attack)` and `_set_current_unit(null)` into the `_on_attack_selected"` function, then you don't need to `yield(HUD, "attack_selected")` *and* you don't need to store the attack in an additional member variable (so `var current_attack` can be removed)

Reply:
Nice, thanks for the feedback I change my code to delete the yield and the current_attack variable.